pmd-core/src/main/java/net/sourceforge/pmd/RuleSetFactory.java

```diff
@@ -587,7 +587,7 @@ private void parseSingleRuleNode(RuleSetReferenceId ruleSetReferenceId, RuleSetB
                 && !isRuleName(ruleElement, ruleSetReferenceId.getRuleName())) {
             return;
         }
-        Rule rule = new RuleFactory().buildRule(ruleElement);
+        Rule rule = new RuleFactory().buildRule(ruleElement, resourceLoader);
```

Comment:
Can you add two constructors in RuleFactory?
* RuleFactory() - default constructor for backwards compatibility
* RuleFactory(ResourceLoader) - the new one to be used here

I think, the resource loader should be a property of the rule factory rather than a method argument for parsing the rule element. In theory, we could reuse the RuleFactory instance for parsing all rules within one ruleset (which we don't do currently).